cpu/esp_common/esp-now/esp_now_netdev.c

```diff
@@ -652,11 +652,11 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
                   data_len + iolist->iol_len, ESP_NOW_MAX_SIZE_RAW);
```

Comment:
Can it be guaranteed that [this dereference above](https://github.com/RIOT-OS/RIOT/pull/11187/files#diff-24495fdc92b602d8c37d024123b775a5R636) does not dereference a NULL pointer?

Reply:
It is guaranteed because this part of the `iol_list` is created and filled by https://github.com/RIOT-OS/RIOT/blob/71204d1d7034b642d310604d0bc9ec4fe634f414/cpu/esp_common/esp-now/esp_now_gnrc.c#L69-L73 as variable on stack before it calls esp_now_netdev::_send with the `iol_list`.

Reply:
But this is only true for GNRC. If another (hypothetical) stack does not necessarily provide this, this can lead to problems. Better document this in the [`esp-now` documentation](https://doc.riot-os.org/esp__now__netdev_8h.html) somewhere.

Reply:
(not a blocker for this PR, as it should go in a separate PR)

Reply:
Though... I just noticed: Having at least a non-zero L2 header should be a requirement for most platforms. Maybe we rather do it the other way around: document that the device *may* need the first header to be of non-zero length, and document the corner cases, where this isn't the case necessarily (e.g. SLIP).